src/installer/tests/TestUtils/TestArtifact.cs

```diff
@@ -57,7 +57,15 @@ public virtual void Dispose()
         {
             if (!PreserveTestRuns() && Directory.Exists(Location))
             {
-                Directory.Delete(Location, true);
+                try
+                {
+                    Directory.Delete(Location, true);
+
+                    // Delete lock file last
+                    Debug.Assert(!Directory.Exists(Location));
+                    var lockPath = Directory.GetParent(Location) + ".lock";
+                    File.Delete(lockPath);
+                } catch {}
```

Comment:
Can we at least write out to console in the case the delete fails (non-error, but get some log)? It may come handy when we're to diagnose the next super-weird behavior in CI.

Reply:
Sure. However, I don't think Delete failing should be a problem -- it just means we can't clean up, which shouldn't be fatal.